From 71aa161979e7bd07c08965da6f0dac32005da6ed Mon Sep 17 00:00:00 2001
From: Guoqing Jiang <guoqing.jiang@suse.com>
Date: Thu, 18 Aug 2022 16:00:41 +0800
Subject: [PATCH] Revert "dmaengine: idxd: add missing callback function to
 support DMA_INTERRUPT"
Patch-mainline: Never, revert stable patch which is added again in sorted section
References: bsn#314
Modified-by-SEL: No


This reverts commit 5e34b4975669451f731ec20b8c0e8c3431d691b3.

Signed-off-by: Guoqing Jiang <guoqing.jiang@suse.com>
---
 drivers/dma/idxd/dma.c | 22 ----------------------
 1 file changed, 22 deletions(-)

diff --git a/drivers/dma/idxd/dma.c b/drivers/dma/idxd/dma.c
index 09ad37bbd98b..d53ce22b4b8f 100644
--- a/drivers/dma/idxd/dma.c
+++ b/drivers/dma/idxd/dma.c
@@ -82,27 +82,6 @@ static inline void idxd_prep_desc_common(struct idxd_wq *wq,
 	hw->int_handle =  wq->vec_ptr;
 }
 
-static struct dma_async_tx_descriptor *
-idxd_dma_prep_interrupt(struct dma_chan *c, unsigned long flags)
-{
-	struct idxd_wq *wq = to_idxd_wq(c);
-	u32 desc_flags;
-	struct idxd_desc *desc;
-
-	if (wq->state != IDXD_WQ_ENABLED)
-		return NULL;
-
-	op_flag_setup(flags, &desc_flags);
-	desc = idxd_alloc_desc(wq, IDXD_OP_BLOCK);
-	if (IS_ERR(desc))
-		return NULL;
-
-	idxd_prep_desc_common(wq, desc->hw, DSA_OPCODE_NOOP,
-			      0, 0, 0, desc->compl_dma, desc_flags);
-	desc->txd.flags = flags;
-	return &desc->txd;
-}
-
 static struct dma_async_tx_descriptor *
 idxd_dma_submit_memcpy(struct dma_chan *c, dma_addr_t dma_dest,
 		       dma_addr_t dma_src, size_t len, unsigned long flags)
@@ -214,7 +193,6 @@ int idxd_register_dma_device(struct idxd_device *idxd)
 	dma_cap_set(DMA_COMPLETION_NO_ORDER, dma->cap_mask);
 	dma->device_release = idxd_dma_release;
 
-	dma->device_prep_dma_interrupt = idxd_dma_prep_interrupt;
 	if (idxd->hw.opcap.bits[0] & IDXD_OPCAP_MEMMOVE) {
 		dma_cap_set(DMA_MEMCPY, dma->cap_mask);
 		dma->device_prep_dma_memcpy = idxd_dma_submit_memcpy;
-- 
2.34.1

